### PR TITLE
feat(material/testing): MatChipHarness getAvatar

### PR DESCRIPTION
**Pull request by tamas-nemeth**
- add `MatChipAvatarHarness`
- add `getAvatar` method to `MatChipHarness`
- make `MatChipHarness` extend `ContentContainerComponentHarness`
  - to allow getting a `MatIconHarness` inside a chip

### CODE REVIEW

File: src/material-experimental/mdc-chips/testing/chip-harness.ts [hunk 1]
```diff
@@ -22,6 +27,11 @@ export class MatChipHarness extends ComponentHarness {
   static with<T extends typeof MatChipHarness>(this: T, options: ChipHarnessFilters = {}):
       HarnessPredicate<InstanceType<T>> {
     return new HarnessPredicate(MatChipHarness, options)
+      .addOption(
+        'hasAvatar',
+        options.hasAvatar,
```
Comment (crisbeto) [inline src/material-experimental/mdc-chips/testing/chip-harness.ts:32]: Is it that common of a use case to be able to query only chips that have an avatar? We don't have anything similar for the other icons that can be projected (e.g. the remove icon).

File: src/material-experimental/mdc-chips/testing/chip-harness.ts [hunk 2]
```diff
@@ -52,4 +62,19 @@ export class MatChipHarness extends ComponentHarness {
   async getRemoveButton(filter: ChipRemoveHarnessFilters = {}): Promise<MatChipRemoveHarness> {
     return this.locatorFor(MatChipRemoveHarness.with(filter))();
   }
+
+  /**
+   * Whether the chip has an avatar
+   */
+   async hasAvatar() {
```
Comment (crisbeto) [inline src/material-experimental/mdc-chips/testing/chip-harness.ts:69]: I don't think that we need this since it can be inferred from `getAvatar`.

Comment (tamas-nemeth) [reply src/material-experimental/mdc-chips/testing/chip-harness.ts]: right, removed

File: src/material-experimental/mdc-chips/testing/chip-harness.ts (hunk 1 above)
Comment (tamas-nemeth) [reply src/material-experimental/mdc-chips/testing/chip-harness.ts]: it's probably not that common. removed

**Review by crisbeto — COMMENTED** [review]
The changes look good, but the API goldens need to be updated. You can run `yarn approve-api chips/testing` to do it.

**Review by mmalerba — APPROVED** [review]
LGTM

**Comment by tamas-nemeth** [comment]
rebased after conflicts

**Pull request merged**